### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by turboMaCk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,342 @@
...
+    controller.set('menuItem', model.decorate('menu-item')); // RETURNS MODEL DECORATED WITH MENU-ITEM DECORATOR
+  }
+});
+```
+
+## Decorating Objects and Collections
+With Clothier it is simple to decorate both objects and collections.
+There are two basic mixins which implements methods for creating decorators instances.
+`ModelMixin` implements `decorate()` method for decorating model instances and helpers for decorating its relationships (see below).
+`RouteMixin` implements `decorateModel()` and `decorateCollection()` methods for both **objects and collections**.
```
Comment (petrjanda) [inline README.md]: Is there anything preventing possible merge of these API calls (`decorateModel()` and `decorateCollection()`) into `decorateModel()` and detect cardinality on the fly?

File: README.md [hunk 2]
```diff
@@ -1,12 +1,342 @@
...
+The difference is that Route methods takes two arguments where first one is model or collection of instances.
+See Api Documentation for more informations.
+
+## Decorating relationships
+`ModelMixin` comes with two additional helper functions for decorating model relationships.
+This helpers takes two arguments ­ *relationKey* and *decoratorName* and return `Ember.computed` which returns decorated relationship.
```
Comment (petrjanda) [inline README.md]: The description "This helpers takes two arguments ­ *relationKey* and *decoratorName* and return ..." contradicts the example bellow - `decorateBelongsTo('author')`.

Also as I've mentioned in https://github.com/GlobalWebIndex/pro/pull/146#discussion_r40257728 the usage of model decorator without a context doesn't sound right. Mandatory decorator name parameter would remove such a confusion (at the potential cost of writing decorator name - which should actually be an indicator of decorator context)

Comment (petrjanda) [inline README.md]: For example if your default decorator would have some logic for search I would do the following:

```
searchableAuthor: decorateBelongsTo('author', 'searchable')
```

and also remove 'decoratedAuthor' which doesn't really help me as a reader to understand what it really means.

File: README.md (hunk 1 above)
Comment (turboMaCk) [inline README.md]: I do not think so. But than it makes sense to me to have also one helper for decorating relations `decorateRelation` for example. We should implement this in utils and reuse it in both mixins.

Comment (turboMaCk) [inline README.md]: Ok, decorating relations using one helper do not seems to be good idea. There are other requirements on computed property for collection / object. Also tests fails for some reason but it is probably related to how computed properties are implemented. Let's keep those helpers separated, at least they will be easily maintainable. I'm OK with merging apis in routeMixin – I will send pull request on this.

Comment (turboMaCk) [inline README.md]: https://github.com/GlobalWebIndex/ember-clothier/pull/11

File: README.md (hunk 2 above)
Comment (turboMaCk) [inline README.md]: Default decorators makes sense to me in context of ember's default naming conventions. The [draper](https://github.com/drapergem/draper) has this to. I think that there should be some real world case where this is useful. For example if you have some model specific helpers for formate `fullName` from `firstName` and `lastName` and `isAdmin` that check if `role === 'admin'`. In this case something like `decorators/user.js` makes sense to me. There basically might be case where someone want to put some logic in default decorator rather then model itself for some reason.

Comment (petrjanda) [inline README.md]: :+1: k, thats good example. Lets see how it goes and revise armed with knowledge from the battlefield.

File: README.md (hunk 1 above)
Comment (petrjanda) [inline README.md]: Excellent!

**Pull request merged**